builder/builder.go

```diff
@@ -0,0 +1,78 @@
+// Package builder provides libraries for the Deis builder.
+//
+// The Deis builder is responsible for packaging Docker images for consumers.
+//
+// The builder/cli package contains command line clients for this library.
+package builder
+
+import (
+	"github.com/Masterminds/cookoo"
+	clog "github.com/Masterminds/cookoo/log"
+	"github.com/deis/deis/builder/sshd"
+
+	"log"
+	"os"
+)
+
+// Return codes that will be sent to the shell.
+const (
+	StatusOk = iota
+	StatusLocalError
+)
+
+// Run starts the Builder service.
+//
+// The Builder service is responsible for setting up the local container
+// environment and then listening for new builds. The main listening service
+// is SSH. Builder listens for new Git commands and then sends those on to
+// Git.
+//
+// Run returns on of the Status* status code constants.
```

Comment:
Thank you very much for writing clear and detailed code comments everywhere. :star: